azurerm/resource_arm_function_app.go

```diff
@@ -402,6 +402,10 @@ func resourceArmFunctionAppRead(d *schema.ResourceData, meta interface{}) error
 
 	appSettingsResp, err := client.ListApplicationSettings(ctx, resGroup, name)
 	if err != nil {
+		if utils.ResponseWasNotFound(appSettingsResp.Response) {
+			log.Printf("[DEBUG] Application Settings of Function App %q (resource group %q) were not found", name, resGroup)
+			return nil
```

Comment:
We should be clearing the ID here:
```suggestion
			d.SetId("")
			return nil
```

Reply:
Got it since the Function app & the settings have one to one mapping, so if the setting is missing it should be marked as not present.